backend/server/src/models/match-model.ts

```diff
@@ -0,0 +1,56 @@
+import mongoose, { Schema, type Types } from "mongoose";
+
+type MatchColor = "red" | "white";
+
+interface MatchPoint {
+  type: string;
```

Comment:
Could we use a union type for this? Using string literals extensively in the codebase might lead to readability and maintenance challenges down the road.